1_fetch/src/get_sb_data.R

```diff
@@ -145,34 +149,62 @@ prep_feature_vars <- function(sb_var_data, sites, retain_vars) {
   }
   
   #handles any duplicated COMIDs
-  data <- data %>% 
-    group_by(COMID) %>%
-    summarise_all(mean)
+  if(!("ID" %in% retain_vars)) {
+    data <- data %>%
+      group_by(COMID) %>%
+      summarise_all(mean) 
+  }
   
   #reclassify land use data for consistency across time periods
-  land_cover <- data %>%
-    select(COMID, 
-           contains("SOHL40"), contains("SOHL50"), contains("SOHL60"),
-           contains("SOHL70"), contains("SOHL80"), contains("SOHL90"), 
-           contains("SOHL00")) %>%
-    pivot_longer(!COMID, names_to = "name", values_to = "value") %>%
-    mutate(unit = str_sub(name, 1, 3), 
-           year = if_else(str_sub(name, 9, 10) == "00", 2000, 
-                          as.numeric(paste0("19", str_sub(name, 9, 10)))), 
-           class = as.numeric(str_sub(name, 12))) %>%
-    mutate(new_class = case_when(class == 1 ~ "WATER", class == 2 ~ "DEVELOPED", 
-                                 class == 3 ~ "FOREST", class == 4 ~ "FOREST", 
-                                 class == 5 ~ "FOREST", class == 6 ~ "MINING", 
-                                 class == 7 ~ "BARREN", class == 8 ~ "FOREST", 
-                                 class == 9 ~ "FOREST", class == 10 ~ "FOREST", 
-                                 class == 11 ~ "GRASSLAND", class == 12 ~ "SHRUBLAND", 
-                                 class == 13 ~ "CROPLAND", class == 14 ~ "HAY/PASTURE", 
-                                 class == 15 ~ "WETLAND", class == 16 ~ "WETLAND", 
-                                 class == 17 ~ "ICE/SNOW")) %>%
-    group_by(COMID, unit, year, new_class) %>%
-    summarise(value = sum(value)) %>%
-    mutate(lc_sum = sum(value)) %>%
-    ungroup()
+  if("ID" %in% retain_vars) {
```

Comment:
Did you check that these ifs for land cover, fire, etc. are needed? I'd expect the resulting tables by COMID to be the same as the tables by GAGES_ID because the data are available for COMIDs.

Reply:
There are more GAGES_ID values than COMID values - without the if statement, the resulting data frames have a different number of rows and mess with the left_join() at the end.

Reply:
yeah. that's what i was expecting too. let me play around with an inner join or something, maybe that is what i need